### ISSUE TITLE
Losing registered devices after doing client.setUser

### ISSUE THREAD

**Issue by pdura** [GetStream/stream-chat-flutter#132]
Hi Guys,
I've got a weird problem with registered devices for a user.

- I launch my application, setup flutter client, get my firebase token and send it to StreamChat API using my server with the python SDK. it's working really well. The token is registered for my user, and notifications are received. 

- Then, as soon as I do 
```
 final client = StreamChat.of(context).client;
 var user = client.setUser(
      User(id: profile.userId),
      profile.streamToken,
 );
```
I do lose my firebase token, even if I don't have any errors on the API call. The response is :
`flutter: (2020-11-10 11:11:26.918156) ℹ️ 🔌 connecting to wss://chat-us-east-1.stream-io-api.com/connect.....
flutter: (2020-11-10 11:11:27.616343) ℹ️ 🔌 received new event: {"type":"health.check","connection_id":"a393fb11-8a87-40bb-aa46-ff0d51d6bc79","cid":"*","me":{"id":"xx","role":"user","roles":[],"created_at":"2020-11-06T16:13:04.560321Z","updated_at":"2020-11-10T10:11:27.507101Z","last_active":"2020-11-10T10:11:27.510877289Z","banned":false,"online":true,"invisible":false,"devices":[],"mutes":[],"channel_mutes":[],"unread_count":0,"total_unread_count":0,"unread_channels":0,"language":"","teams":["xx"],"name":"xx","image":xx"},"created_at":"2020-11-10T10:11:27.51291756Z"}`

The weird thing is that teams setup is not lost, only devices.
Thanks a lot for your help.

Flutter Doctor 
[✓] Flutter (Channel stable, 1.22.3, on Mac OS X 10.15.7 19H2, locale fr-FR)
    • Flutter version 1.22.3 at /Users/patrick/project/flutter
    • Framework revision 8874f21e79 (12 days ago), 2020-10-29 14:14:35 -0700
    • Engine revision a1440ca392
    • Dart version 2.10.3

stream_chat_flutter: ^0.2.11+1

gz#7241

**Comment by shodgetts** [comment]
Agent comment from Séni Gueye in <a href='https://getstream.zendesk.com/agent/tickets/7241'>Zendesk ticket #7241</a>:<br><br>Hi,

Front-end tokens are not returned in setUser() response, or users objects either.
Also, I see you're referring to a firebase token.
Can you tell me more about it? Is it a JWT token? Could you please share the code where generating the token using Python server-side SDK?

Thanks in advanve.
Best regards,

**Comment by pdura** [comment]
Hi,
Yes sorry I haven't put that code, and I might miss details. Code below.
```
python : stream-chat==1.7.0


import stream_chat
from flask import current_app
import datetime

class GTStreamChat(object):

    def __init__(cls):
        stream_key = current_app.config.get("STREAM_API_KEY").encode('utf-8')
        stream_secret = current_app.config.get("STREAM_API_SECRET").encode('utf-8')
        cls.chat_client = stream_chat.StreamChat(api_key=stream_key, api_secret=stream_secret)

    def create_token(self, user_id):
        return self.chat_client.create_token(str(user_id), exp=datetime.datetime.utcnow() + datetime.timedelta(hours=1))

    def add_device_token(self, user_id, token):
        return self.chat_client.add_device(token, 'firebase', str(user_id))

    def remove_device_token(self, user_id, token):
        return self.chat_client.delete_device(token, str(user_id))

    def get_devices(self, user_id):
        return self.chat_client.get_devices(str(user_id))
        
        
```

Explanations :

1. Server side, I'm generating the User JWT token using `create_token` below. That's for the auth part and it's working like a charm.

2. Flutter side, I launch the app, setup client, and get my device token for notifications and push it to my own server.
That's also working well.
`
final FirebaseMessaging _firebaseMessaging = FirebaseMessaging();
 _firebaseMessaging.requestNotificationPermissions(
        const IosNotificationSettings(
            sound: true, badge: true, alert: true, provisional: false));
    _firebaseMessaging.onIosSettingsRegistered
        .listen((IosNotificationSettings settings) {
      print("Settings registered: $settings");
    });
    _firebaseMessaging.getToken().then((String token) async {
      assert(token != null);
      print("Push Messaging token: $token");
      final auth = await Provider.of<Auth>(context, listen: false);
      GTApi(token: await auth.token).registerPushNotificationToken(token);
    });
`

3. Server side, I register my device using `add_device_token`. If I check using `get_devices` the new device token do appears on my list.

4. Flutter side, I run `client.setUser(User(userId), token)`.

5. Server side, I check using `get_devices`, I've got an empty list. no more devices registered.

FYI I just moved to `client.setUserWithProvider` and this one seems to keep my devices registered on stream.
I still need to run a few tests though to make sure of that.

Regards,
Patrick

**Comment by imtoori** [comment]
Hi @pdura 
I'm trying to understand what's going on here
A weird thing is that now it's working using `client.setUserWithProvider`, because this does the same API calls as `client.setUser` does
Do you have any update on your side?

**Comment by pdura** [comment]
Ok guys, I think I found how I'm losing my devices token.
Sounds like it's linked to connecting a client without the correct base_url.

I can reproduce my problem doing the following 
- On Stream, Create a chat app located in dublin
- On flutter, setup a client with the base_url = 'https://chat-proxy-dublin.stream-io-api.com'.
With that one, register a device token for a user. The device token on stream => all good.
- to do my tests, I setup a react chat from your tutorial (https://getstream.io/chat/react-chat/tutorial/)
- Create the client WITHOUT the base url.
- Check stream tokens. they are gone.

I just did that a few time, and every time I lose my tokens.
If I setup the base url on the react client, no problems, I don't lose my tokens any more.
